### PR TITLE
Add LLM Provider interface and Gemini module

### PR DESCRIPTION
**Pull request by sydp**
Adds:
* an interface/manager for LLM Providers
* a Gemini provider module
* ~~a VertexAI provider module~~
* ~~a Ollama provider module~~
* updates /data/config.json with example configs for LLM providers
* refactor of LLMProcessorBase

### CODE REVIEW

**Comment by sydp** [comment]
Removed some modules to keep the PR to a managable size and will send these separately.

File: dftimewolf/lib/processors/llmproviders/gemini.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    history_patched = []
+    for content in self.chat_session.history:
+      if not content.parts:
+        content.parts = [genai.types.content_types.to_part(patch_content)]
+      history_patched.append(content)
+    self.chat_session.history = history_patched
+
+  def AskGemini(self, prompt: str, model: str) -> str:
+    """Ask Gemini."""
```
Comment (ramo-j) [inline dftimewolf/lib/processors/llmproviders/gemini.py:191]: nit: Docstring args

**Review by ramo-j — DISMISSED** [review]
LGTM, just a couple of small nits.

File: dftimewolf/lib/processors/llmproviders/gemini.py (hunk 1 above)
Comment (sydp) [reply dftimewolf/lib/processors/llmproviders/gemini.py]: Done

**Review by ramo-j — APPROVED** [review]

**Pull request merged**